tests/test_rechunk.py

```diff
@@ -41,6 +41,130 @@ def test_invalid_executor():
+    )
+    return ds.chunk({"time": -1})
+
+
+@pytest.mark.parametrize(
+    "target_chunks,expected",
+    [
+        pytest.param(
+            dict(lon=10),
+            dict(aaa=(10, 180, 365), lon=(10,), lat=(180,), time=(365,)),
+            id="just lon chunk",
+        ),
+        pytest.param(
+            dict(lat=10),
+            dict(aaa=(360, 10, 365), lon=(360,), lat=(10,), time=(365,)),
+            id="just lat chunk",
+        ),
+        pytest.param(
+            dict(time=10),
+            dict(aaa=(360, 180, 10), lon=(360,), lat=(180,), time=(10,)),
+            id="just time chunk",
+        ),
+        pytest.param(
+            dict(lon=10, lat=10, time=10),
+            dict(aaa=(10, 10, 10), lon=(10,), lat=(10,), time=(10,)),
+            id="all dimensions - equal chunks",
+        ),
+        pytest.param(
+            dict(lon=10, lat=20, time=30),
+            dict(aaa=(10, 20, 30), lon=(10,), lat=(20,), time=(30,)),
+            id="all dimensions - different chunks",
+        ),
+        pytest.param(
+            dict(lon=1000),
+            dict(aaa=(360, 180, 365), lon=(360,), lat=(180,), time=(365,)),
+            id="lon chunk greater than size",
+        ),
+        pytest.param(
+            dict(lat=1000),
+            dict(aaa=(360, 180, 365), lon=(360,), lat=(180,), time=(365,)),
+            id="lat chunk greater than size",
+        ),
+        pytest.param(
+            dict(time=1000),
+            dict(aaa=(360, 180, 365), lon=(360,), lat=(180,), time=(365,)),
+            id="time chunk greater than size",
+        ),
+        pytest.param(
+            dict(lon=1000, lat=1000, time=1000),
+            dict(aaa=(360, 180, 365), lon=(360,), lat=(180,), time=(365,)),
+            id="all chunks greater than size",
+        ),
+    ],
+)
+def test_parse_target_chunks_from_dim_chunks(
+    chunk_ds: xarray.Dataset, target_chunks, expected
+) -> None:
+    result = api.parse_target_chunks_from_dim_chunks(
+        ds=chunk_ds, target_chunks=target_chunks
+    )
+    assert expected == result
+
+
+@pytest.mark.parametrize(
+    "dask_chunks, dim, target_chunks, expected",
+    [
+        pytest.param(
+            False, "lon", dict(lon=10), 10, id="small lon chunks numpy array",
+        ),
+        pytest.param(False, "lon", dict(lon=10), 10, id="small lon chunks dask array",),
+        pytest.param(False, "time", dict(time=400), 365, id="time chunks exceed len",),
+        pytest.param(False, "time", dict(time=-1), 365, id="negative time chunks",),
+        pytest.param(
+            {"time": 1},
+            "time",
+            dict(time=-1),
+            365,
+            id="negative time chunks dask array",
+        ),
+        pytest.param(
+            {"time": 5}, "time", dict(), 5, id="preserve small time chunks dask array",
+        ),
+        pytest.param(
+            False, "lon", dict(), 360, id="preserve full length chunk numpy array",
+        ),
+    ],
```

Comment:
This is purely a personal preference, so feel free to ignore it. Exceeding the line length limit does not unconditionally mean that the code is less readable. `pytest.param()` lines tend to often be longer than what makes sense for normal code. Therefore I usually keep them in a single line and ask `black` to ignore the relevant lines by adding `fmt: off` and `fmt: on` comments. E.g.

``` python
@pytest.mark.parametrize(
    "dask_chunks, dim, target_chunks, expected",
    [
        # fmt: off
        pytest.param(False, "lon", dict(lon=10), 10, id="small lon chunks numpy array",),
        pytest.param(False, "lon", dict(lon=10), 10, id="small lon chunks dask array",),
        pytest.param(False, "time", dict(time=400), 365, id="time chunks exceed len",),
        pytest.param({"time": 1}, "time", dict(time=-1), 365, id="negative time chunks dask array",),
        # fmt: on
    ]
)
def test_get_dim_chunk(dask_chunks, chunk_ds, dim, target_chunks, expected):
    ...
```

Reply:
I didnt even know this was an option. Happy to implement.

Reply:
@pmav99 this seems to work with black, but not flake8. Is there a similar workaround? Otherwise I would probably return this to the original state to move this along if that is ok?